### PR TITLE
[ant-1860] Rename  outer_loop  --> adequacy_criterion

### PR DESCRIPTION
**Pull request by a-zakir**

### CODE REVIEW

File: src/cpp/benders/adequacy_criterion/AdequacyCriterion.cpp [hunk 1]
```diff
@@ -47,27 +46,27 @@ void OuterLoop::Run() {
...
   benders_->free();
 }
 
-void OuterLoop::PrintLog() {
+void AdequacyCriterion::PrintLog() {
   std::ostringstream msg;
   auto logger = benders_->_logger;
   logger->PrintIterationSeparatorBegin();
-  msg << "*** Outer loop: " << benders_->GetBendersRunNumber();
+  msg << "*** Adequacy Criterion: " << benders_->GetBendersRunNumber();
```
Comment (tbittar) [inline src/cpp/benders/adequacy_criterion/AdequacyCriterion.cpp]: `Adequacy criterion loop`  maybe ? As we print the number of iterations of a loop here

File: src/cpp/benders/benders_core/AdequacyCriterionInputDataReader.cpp [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    throw AdequacyCriterionInputFileIsEmpty(
+        PrefixMessage(LogUtils::LOGLEVEL::FATAL, "Adequacy Criterion"),
+        err_msg.str(), LOGLOCATION);
+  }
+
+  return yaml_content.as<AdequacyCriterionInputData>();
+}
+
+/*
+# critère d'arrêt de l'algo
```
Comment (tbittar) [inline src/cpp/benders/benders_core/AdequacyCriterionInputDataReader.cpp]: Is this comment useful in the code ?

**Review by tbittar — CHANGES_REQUESTED** [review]

File: src/cpp/benders/adequacy_criterion/AdequacyCriterion.cpp (hunk 1 above)
Comment (a-zakir) [reply src/cpp/benders/adequacy_criterion/AdequacyCriterion.cpp]: updated

File: src/cpp/benders/benders_core/AdequacyCriterionInputDataReader.cpp (hunk 2 above)
Comment (a-zakir) [reply src/cpp/benders/benders_core/AdequacyCriterionInputDataReader.cpp]: no

**Pull request closed without merging**